xbmc/video/VideoInfoScanner.cpp

```diff
@@ -237,201 +246,159 @@ namespace VIDEO
-    { // update the hash either way - we may have changed the hash to a fast version
-      m_database.SetPathHash(strDirectory, hash);
+      m_handle->SetTitle(g_localizeStrings.Get(20415));  //"Scanning for new content"
     }
 
-    if (m_handle)
-      OnDirectoryScanned(strDirectory);
 
-    for (int i = 0; i < items.Size(); ++i)
-    {
-      CFileItemPtr pItem = items[i];
+    // Calculate total items count to display a proper progress bar
+    int totalItems = 0;
+    for (const auto& pathContent : _pathsContent) {
+      totalItems += pathContent.items.Size();
+    }
 
-      if (m_bStop)
-        break;
+    int itemsChecked = 0;
+    for (const auto& pathContent : _pathsContent) {
+      const CFileItemList& remoteDirectoryItems = pathContent.items;
 
-      // add video extras to library
-      if (foundSomething && !m_ignoreVideoExtras && pItem->IsVideoExtras())
+      for (int i = 0; i < remoteDirectoryItems.Size(); ++i)
       {
-        if (AddVideoExtras(items, content, pItem->GetPath()))
-        {
-          CLog::Log(LOGDEBUG, "VideoInfoScanner: Finished adding video extras from dir {}",
-                    CURL::GetRedacted(pItem->GetPath()));
+        CFileItemPtr remoteItem = remoteDirectoryItems[i];
+        std::string remoteItemPath = remoteItem->GetPath();
+
+        // Update progress bar
+        if (m_handle) {
+          itemsChecked++;
+          m_handle->SetPercentage(itemsChecked*100.f/totalItems);
+          m_handle->SetText(remoteItem->GetMovieName(true));
         }
 
-        // no further processing required
-        continue;
-      }
+        // Support scan cancellation
+        if (m_bStop)
+          return !m_bStop;
 
-      // if we have a directory item (non-playlist) we then recurse into that folder
-      // do not recurse for tv shows - we have already looked recursively for episodes
-      if (pItem->m_bIsFolder && !pItem->IsParentFolder() && !pItem->IsPlayList() && settings.recurse > 0 && content != CONTENT_TVSHOWS)
-      {
-        if (!DoScan(pItem->GetPath()))
+
+        if( (pathContent.contentType_AsDefinedInScraper == CONTENT_TVSHOWS && remoteItem->m_bIsFolder) //For TV shows, we only scan folders
+             ||
+             !remoteItem->m_bIsFolder)  // if item is not a folder, it's a file we want to scan
         {
-          m_bStop = true;
+          // Check if video file is already in DB:
+          bool isVideoAlreadyInDB = false;
+          switch (pathContent.contentType_AsDefinedInScraper)
+          {
+            case CONTENT_MOVIES:
+              isVideoAlreadyInDB = m_database.HasMovieInfo(remoteItemPath);
+              break;
+            case CONTENT_MUSICVIDEOS:
+              isVideoAlreadyInDB = m_database.HasMusicVideoInfo(remoteItemPath);
+              break;
+            case CONTENT_TVSHOWS:
+              isVideoAlreadyInDB = m_database.HasTvShowInfo(remoteItemPath);
+              break;
+          }
```

Comment:
I would insert a `default` label to indicate that all intended value are handled here. as far as I know there are more types than only those handled here. Compiler should even generate a warning that you do not handle all possible values here?

Reply:
Thanks for the comment.
I thought about this but found no action I would do in such "default" case.

If it's really mandatory to have one I would write:
```
default:
  isVideoAlreadyInDB = false;
  break;
```

Which does not make sense as isVideoAlreadyInDB is by default set to false anyway.
Not sure what I should change here                      _(and I did not check compiler's output)_

I would prefer to leave it like that, but feel free to give more orientations.